sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/assessment-workflow-enqueuer.scheduler.ts

```diff
@@ -35,39 +35,33 @@ export class AssessmentWorkflowEnqueuerScheduler extends BaseScheduler<void> {
    * @returns processing result.
    */
   @Process()
-  async enqueueAssessmentOperations(
-    job: Job<void>,
-  ): Promise<QueueProcessSummaryResult> {
-    const summary = new QueueProcessSummary({
-      appLogger: this.logger,
-      jobLogger: job,
-    });
-    await summary.info(
-      "Checking application assessments to be queued for start.",
-    );
-    // Process summary to be populated by the enqueueStartAssessmentWorkflows.
-    // In case an unexpected error happen the finally block will still be able to
-    // output the partial information captured by the processSummary.
+  async enqueueAssessmentOperations(job: Job<void>): Promise<string[]> {
     const processSummary = new ProcessSummary();
     try {
+      processSummary.info(
+        "Checking application assessments to be queued for start.",
+      );
+      // Process summary to be populated by the enqueueStartAssessmentWorkflows.
+      // In case an unexpected error happen the finally block will still be able to
+      // output the partial information captured by the processSummary.
+      const serviceProcessSummary = new ProcessSummary();
+      processSummary.children(serviceProcessSummary);
```

Comment:
Let me know if I am missing something. this part can be done inside process summary class constructor right?

Reply:
Sorry, I am not following the comment. Do you mean the association of a child process?
Here we have a parent log and the creating of a child log to capture the service-related logs. I am not sure what could be done inside the `ProcessSummary` constructor.

Reply:
I get it now. it is for the first level of indentation. 
![image](https://github.com/bcgov/SIMS/assets/54600590/57d16688-9ac4-49cc-aaee-e680487c4037)

My understanding is better now.

With that in mind, just wondering if service process summary could be retuned by a getter in the class like this.  So that we can avoid doing this assignment at processor level.

```ts
await this.workflowEnqueuerService.enqueueStartAssessmentWorkflows(
        processSummary.getServiceProcessSummary(),
      );
```

Reply:
The idea of having it created at the processor lever and passing it as a reference is to allow access to it in case some unhandled exception happens inside the service. Having it at the processor level allows us to dump any information added to it in the final block. Does it make sense?